group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/EventAccumulator.java

```diff
@@ -137,31 +136,43 @@ public void add(T event) throws RejectedExecutionException {
     }
 
     /**
-     * Returns the next {{@link Event}} available. This method block indefinitely until
-     * one event is ready or the accumulator is closed.
+     * Returns the next {{@link Event}} available or null if no event is
+     * available.
      *
-     * @return The next event.
+     * @return The next event available or null.
      */
     public T poll() {
-        return poll(Long.MAX_VALUE, TimeUnit.SECONDS);
+        lock.lock();
+        try {
+            K key = randomKey();
+            if (key == null) return null;
+
+            Queue<T> queue = queues.get(key);
+            T event = queue.poll();
+
+            if (queue.isEmpty()) queues.remove(key);
+            inflightKeys.add(key);
+            size--;
+
+            return event;
+        } finally {
+            lock.unlock();
+        }
     }
 
     /**
-     * Returns the next {{@link Event}} available. This method blocks for the provided
-     * time and returns null of not event is available.
+     * Returns the next {{@link Event}} available. This method blocks until an
+     * event is available or the thread is interrupted.
```

Comment:
I noticed we just catch interrupted exceptions. Is that correct? Or should we adjust the comment to say or the accumulator is closed.